internal/ccsmp/cgo_helpers.go

```diff
@@ -53,3 +55,17 @@ func ToCArray(arr []string, nullTerminated bool) (cArray **C.char, freeArray fun
 	}
 	return (**C.char)(unsafe.Pointer(&cArr[0])), freeFunction
 }
+
+func GenerateTestSolClientErrorInfoWrapper(returnCode SolClientReturnCode, subCode SolClientSubCode, responseCode SolClientResponseCode, errorInfo string) *SolClientErrorInfoWrapper {
```

Comment:
Perhaps this should be called NewSolClientErrorInfoWrapper?

Reply:
This is used to avoid code duplication in some of the unit tests, look for the references to this function for an example. We could rename it if you want. The reason it can't be defined in one of the unit test files is because I got a compiler warning saying I wasn't permitted to use cgo in the unit test files. So, I had to write this function in a separate file under the `internal/ccsmp` directory where cgo is accessible.

Reply:
I added the word `Test` to the function name since it is only used by the tests, and because it is generating an error info representation. I feel like just calling it `NewSolClientErrorInfoWrapper` could mislead developers into thinking its a regular constructor that can be used anywhere instead of a test helper, since devs should not be generating the last error info from ccsmp outside of test scenarios. What are your thoughts?

Reply:
Maybe we should be using that constructor or we need to complicate the unit test source. My vote is for adding the new constructor method to make the error info struct.

As a constructor it should not have the word `Generate` in it. It Should be `New`. As its not in test code it must not have the word `Test` in it. Though if its internal maybe have the word `Internal`. If you feel needing the separation of the internal use or not. I'd also recommend adding comments to the internal construction independent of this function in the api as for optimization purposes do not use the constructor method. But in tests optimization is not much of a concern. And maybe a description comment on the New method should include a "standard" for making errorInfoWrapper and that is not optimized.